applications/temporalgrid-demo/src/App.js

```diff
@@ -113,50 +176,67 @@ export default function App() {
           serverSideFilter: undefined,
           // serverSideFilter: `vesselid IN ('ddef384a3-330b-0511-5c1d-6f8ed78de0ca')`,
           updateColorRampOnTimeChange: true,
-          zoom: viewport.zoom,
+          // zoom: viewport.zoom,
           fetchStats: true
         })
       }
-
+    console.log(generators)
     return generators
   },
-    [animated, viewport, showBasemap, debug, debugLabels, tilesets, geomTypeMode, isPlaying, combinationMode]
+    [animated, showBasemap, debug, debugLabels, tilesets, geomTypeMode, isPlaying, combinationMode]
   );
 
-  // TODO switch between debounced/immediate/throttled time when using animated
-  const { style } = useLayerComposer(layers, (animated) ? time: debouncedTime)
+  const [mapRef, setMapRef] = useState(null)
+
+  const [clickedEvent, setClickedEvent] = useState(null)
+  const clickCallback = useCallback((event) => {  
+    // probably dispatch a redux action here or whatever
+    setClickedEvent(event)
+    // TODO
+    // if (feature.popupCallbackURL) {
+    //   // dispatchEvent(...)
+    // }
+  }, [])
+  const closePopup = useCallback(() => {
+    setClickedEvent(null)
+  }, [])
+
+  const [hoveredEvent, setHoveredEvent] = useState(null)
+  const hoverCallback = useCallback((event) => {
+    setHoveredEvent(event)
+  }, [])
 
-  const mapRef = useRef(null)
-  if (mapRef && mapRef.current) {
-    mapRef.current.getMap().showTileBoundaries = debug
-    mapRef.current.getMap().on('idle', () =>  {
+  const { onMapClick, onMapHover } = useMapInteraction(clickCallback, hoverCallback, mapRef)
```

Comment:
I think we could split by callback in the same way we did for the useTooltip for the different event, so in case we only want hover we don't have to send clickCallback as undefined